<a href="https://colab.research.google.com/github/Kiran-Shehzadi/flyrank-ML_Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Kiran-Shehzadi/flyrank-ML_Internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [8]:
!git clone https://github.com/Kiran-Shehzadi/flyrank-ML_Internship.git

fatal: destination path 'flyrank-ML_Internship' already exists and is not an empty directory.


## My Provisional Lane

I have chosen the **Content Refresh Prioritization** lane.

The objective is to help an SEO team...

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Problem Framing

### Decision

Which pages should an SEO team prioritize for a content refresh?

### Who acts on the output?

The primary user is an SEO specialist or content editor.

They would use the ranked list of pages to decide which pages should be reviewed and updated first.

### Action

For the highest-priority pages, the SEO team may:

- Update outdated content
- Improve keyword optimization
- Refresh statistics and references
- Improve internal linking
- Rewrite titles or meta descriptions

### Cost of a Wrong Recommendation

If a page is incorrectly identified as needing a refresh, valuable editor time is wasted on pages that may already be performing well.

If an important declining page is missed, it may continue losing impressions, clicks, and organic traffic before it receives attention.

Because both types of errors have different consequences, the model should support prioritization rather than make automatic decisions.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("flyrank-ML_Internship/data/raw/content_refresh_anonymized.csv")
# Basic statistics
print("Number of Pages:", len(df))
print("Number of Features:", len(df.columns))

print("\nAverage CTR:", round(df["ctr"].mean(), 4))
print("Average Position:", round(df["avg_position"].mean(), 2))
print("Average Content Age (days):", round(df["content_age_days"].mean(), 2))

print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())

Number of Pages: 30000
Number of Features: 44

Average CTR: 0.5107
Average Position: 16.34
Average Content Age (days): 256.17

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## Quick Look at the Data

The starter dataset contains **30,000 pages** and **44 features**.

From the initial exploration:

- The average CTR is **0.5107**.
- The average search position is **16.34**.
- The average content age is **256.17** days.

The dataset contains 16,262 pages with a downward trend, indicating that a substantial portion of pages may require review. Combined with variation in CTR, average position, and content age, this supports investigating a method to prioritize pages for content refresh. These observations suggest that prioritizing pages for review is a meaningful decision-support problem, making the Content Refresh Prioritization lane a suitable choice for this project.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What I Can and Can't Claim

This project is intended to provide **decision support** for SEO specialists by identifying pages that may deserve a content refresh based on observed historical performance patterns.

The results will describe **observed** and **directional** relationships within the available dataset. They may help prioritize pages for review but do not prove that any feature causes changes in search performance.

This project does **not** predict Google's ranking algorithm, establish causal relationships, or guarantee future page performance. Any recommendations produced by the model should be used alongside human judgment and domain expertise.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.